geomstats/test_cases/geometry/discrete_curves.py

```diff
@@ -1,13 +1,92 @@
 import pytest
 
 import geomstats.backend as gs
+from geomstats.numerics.finite_differences import forward_difference
 from geomstats.test.random import get_random_times
 from geomstats.test.vectorization import generate_vectorization_data
 from geomstats.test_cases.geometry.fiber_bundle import FiberBundleTestCase
 from geomstats.test_cases.geometry.nfold_manifold import NFoldManifoldTestCase
+from geomstats.test_cases.geometry.pullback_metric import PullbackDiffeoMetricTestCase
 from geomstats.vectorization import get_batch_shape
 
 
+def elastic_distance(space, point_a, point_b, a=1.0, b=0.5):
```

Comment:
Should this overwrite `elastic_metric.dist` ? I guess this depends on the answer to the question: which one is faster 1) our current implementation of `dist` that uses our classes and the diffeo abstraction, or 2) this function?

Reply:
Spot on!

A quick test has shown `elastic_distance` to be faster (though both are fast).
Still, I have a refactoring in mind for `NFoldMetric` that is likely to put both on pair or even make the current one faster (it will take advantage of the fact most operations are vectorized for multiple batch dimensions).

I suggest to merge as is, and check this again after the refactoring I'm suggesting for the `NFoldMetric`.


(We can actually make `elastic_distance` even faster by computing the length directly from the velocity (since we use the point, we need to do finite differences again inside).)

Reply:
We'll check this again later @ninamiolane. I'm merging now as it is useful.